Update DG with usecases (#157)
AY2324S1-CS2103T-T13-2/tp: docs/DeveloperGuide.md

(For all use cases below, the **System** is `CampusConnect` and the **Actor** is a `NUS student who stays in campus`, unless specified otherwise)

> **Use case: UC1 - Add new contact**

Reviewer: this one the second step of the use case is `System shows a list of all contacts.` should change to `System adds the new contact into its system.`

* 1a. System detects an error in the entered data.
    * 1a1. System shows an error message.
    * 1a2. User enters a new add request. <br>
      Steps 1a1- 1a2 are repeated until the data entered is correct.<br>
      Use case resumes from step 2. <br>

> **Use case: UC2 - List all contacts**

Reviewer: under this use case, just realized that step 2 is `System adds the new contact into its system`. should change to `System displays all the contacts saved in its system`.

Author: Done!

* 2a. System detects an invalid index entered.
  * 2a1. System shows an error message.
  * 2a2. User enters a new delete request. <br>
  Steps 2a1- 2a2 are repeated until the data entered is correct.<br>
  Use case resumes from step 3. <br>

**Use Case UC6 - Add note**

**MSS**
1. User <ins>lists all contacts (UC2).</ins>

> 2. User enters an index to add a note associated with a contact

Reviewer: here, this sounds like user just enters index for this command.

```suggestion
2. User enters an index of the specified contact alongside with the notes to be added.
```

3. System adds the note to the contact.<br>
Use case ends.

**Extensions**
* 2a. System detects an invalid index entered.
  * 2a1. System shows an error message.

>   * 2a2. User tries to add a note to a contact again.<br>

Reviewer: should we standardize this with the above UCs?

```suggestion
  * 2a2. User enters a new addnote request.
```

  Steps 2a1- 2a2 are repeated until the data entered is correct. <br>
  Use case resumes from step 3.

**Use Case UC7 - Record payment**

**MSS**
1. User <ins>lists all contacts (UC2).</ins>

> 2. User enters an index to record an amount of money paid to a contact.

Reviewer: afaik, it is not necessarily `paid` but also, there is a concept of `owe`. perhaps we can combine them into this single use case?

```suggestion
2. User enters an index of the specified contact alongside with the amount of money paid/owed to that contact.
```

> 3. System records money paid to the contact.

Reviewer: to change to `paid/owed` if decide to combine into a single use case

4. System displays the contact with an indication of the money owed.<br>
Use case ends.

**Extensions**
* 2a. System detects an invalid index entered.
  * 2a1. System shows an error message.

>   * 2a2. User tries to record a payment to a contact again.<br>

Reviewer: ditto, see above comments for standardization

  Steps 2a1- 2a2 are repeated until the data entered is correct. <br>
  Use case resumes from step 3.
* 2b. System detects an invalid payment amount entered.
  * 2b1. System shows an error message.
  * 2b2. User tries to record a payment to a contact again.<br>
  Steps 2b1- 2b2 are repeated until the data entered is correct. <br>
  Use case resumes from step 3.

**Use Case UC8 - Search for contact**

**MSS**
1. User enters a search criteria to find a contact.
2. System displays all contacts that match the criteria.<br>
Use case ends.

**Extensions**
* 1a. System detects an invalid search criteria.
  * 1a1. System shows an error message.

>   * 1a2. User tries to enter a search criteria to find a contact.<br>

Reviewer: ditto, see above comments
